### ISSUE TITLE
Release v2.0.0?

### ISSUE THREAD

**Issue by fran-worley** [offirgolan/ember-light-table#739]
@alexander-alvarez what are your thoughts on releasing a full version of v2.x once the errors in v3.4 are fixed?

The current v2.x branch provides deprecation free support for ember v3.4 - v3.16 with basic angle bracket syntax support. This branch has been in development for years so it would be good to get a full version released!

We could then start work on a v3 which could make use of glimmer components etc. which are only available in later versions of ember. 

Once that is done we could look to adding some of the features originally planned for v2 which never made it

**Comment by alexander-alvarez** [comment]
Sounds like a great idea. Did we ever get your publish rights?

**Comment by fran-worley** [comment]
@alexander-alvarez I've just checked ruby-gems and I don't have rights - can you give them to me?

**Comment by alexander-alvarez** [comment]
what's your npm username?

**Comment by fran-worley** [comment]
fworley - just created an account!

**Comment by Gorzas** [comment]
@fran-worley Can I help with the roadmap of ember-light-table?

**Comment by fran-worley** [comment]
@Gorzas yes please!! If we can fix the failing tests on the 2.x branch for ember 3.4 we can do a full release of 2.0.0.

Otherwise do you want to start a v3 branch (off V2.x and start on octane optimising? We can begin with basic admin - dropping support for earlier versions of ember, converting to native classes with decorators etc. Then we should start a new issue to review the component API as it will need to change to enable us to make best use of angle bracket syntax.

**Comment by Gorzas** [comment]
Ok, I'm going to take a look on this failing tests first :).

**Comment by Gorzas** [comment]
@fran-worley I've been debugging the problem with ember-viewport. It seems that, with a different Ember version, it uses a different resolver. Don't know if it's really a problem with Ember-viewport but it doesn't seem it. No clue how to fix it right now, I'm going to keep seeking a solution to this problem.

**Comment by fran-worley** [comment]
@Gorzas any chance you could throw up a reproduction app in 3.4 with ELT at 2.x branch and see if it still happens? If not I'm happy to release beta6 on the basis the issue is with our test environment and not ELT or InViewport. 
We can then work on an internal fix in our own time.

**Comment by Gorzas** [comment]
Ok, I'll try it tomorrow, I may have a couple of free hours to do it.

Meanwhile, I've been redirected to [this issue](https://github.com/stefanpenner/ember-strict-resolver/issues/22) but I still don't know the differences between the different  Ember Resolvers or where are they use it.

**Comment by Gorzas** [comment]
@fran-worley I tried it with the dummy app and it doesn't work. Let me take this some more time if I can fix it. Sorry I couldn't try it before 😅 .

**Comment by Gorzas** [comment]
How do you see dropping the support to older versions of Ember? I've spent a lot of time debugging the addon and I still don't see how to fix the problem. I've seen that the version we use of `ember-in-viewport` uses `v3.12` and we want to be compatible with `v3.4`. What do you think? People with older versions of Ember will have `v1.x`.

**Comment by fran-worley** [comment]
@Gorzas let's drop it. I've seen plenty of addons which have done far more brutal version support culls recently in the interest of progress.

**Comment by Gorzas** [comment]
I've created a PR with the changes. It also drops support for Ember 3.5 :(.

**Comment by corrspt** [comment]
Hey

I'm not entirely sure I can be of help, but I'm using Light Table 2.0.0-beta 4 and having trouble upgrading to Ember 3.16 (I'm on 3.13, making my way to 3.15) and Light table is causing the errors starting with 3.16. Anyone using light table with 3.16+?

I'd like to contribute if possible (even if just making things run with 3.16)

**Comment by maxwondercorn** [comment]
@corrspt I'm using ELT with 3.16 with Ember components, not glimmer components. You might want to open an issue with the errors you are seeing or ask in tables topic in the Discord server

**Comment by Gorzas** [comment]
@fran-worley I've seen that some checks have failed, is there anything I should look out?

Let me know what else it's needed to release version 2.0 :smile: .

**Comment by corrspt** [comment]
@maxwondercorn thanks for the comment. In my next upgrade window (soon I hope) I'll do just that

**Comment by fran-worley** [comment]
@Gorzas I have no idea what the problem is with V2... it got very late... 

I think I just need to revert back to before I started looking at it and then release with updated docs. I suggest anyone on ember >3.16 uses the new 3.x branch

**Comment by Gorzas** [comment]
I have time to give some attention to the library. You were talking about going directly to version 3.x with `ember-light-tablè, am I right?

I'm going to do some PRs this pair of weeks in case I can be of any help.

**Comment by fran-worley** [comment]
That would be awesome @Gorzas . I can't see the point in fixing random failing tests with old versions of ember when most addons have dropped support anyway.

**Comment by rwwagner90** [comment]
@fran-worley @Gorzas what's the status on the 2-x and 3-x branches? Can we get all the latest stuff merged into master? I think we should stop supporting older Ember versions and just get all the new stuff in, so we can continue to move forward.

**Comment by Gorzas** [comment]
@rwwagner90 I think the version 2-x has been rejected as it has arrived too late. Lately, I've been using the current beta 3-x without having problems. It needs a migration to Octane and start using native classes to finish the release. I don't know if it also requires some docs update.

**Comment by rwwagner90** [comment]
@Gorzas what do you mean it arrived too late? I just want to get all the latest and best changes into master, we shouldn't have all these different branches.

**Comment by Gorzas** [comment]
@rwwagner90 I mean that it already exists a working version 3, so I think the 2-x version could be avoided. I don't know if there are commits in the branch that aren't in the 3-x branch.

**Comment by rwwagner90** [comment]
@Gorzas gotcha. Who would best know the status of things? @fran-worley do you know if we could just merge 3-x into master and delete 2-x?

**Comment by maxwondercorn** [comment]
Looking through the history, 3-x is branched from 2-x.  The first commit was 91a4fa4

**Comment by rwwagner90** [comment]
Would love to hear from some folks who created the 2-x and 3-x branches to make sure basically replacing master with one of them is okay.

**Comment by maxwondercorn** [comment]
@fran-worley @Gorzas @alexander-alvarez anyone? see ^^^^^^

**Comment by rwwagner90** [comment]
If we don't hear from anyone, I think I'll likely just merge 3-x into master and delete 2-x and 3-x branches so we can move forward on this project.

**Comment by fran-worley** [comment]
@rwwagner90 merge away, my time was always going to be very limited on this and I felt awkward making decisions without support from longer standing maintainers.

Given how far ember has moved on, my preference would be to set version 3 as minimum of v3.20. If I can help do let me know but as I say, my time is quite limited.

**Comment by rwwagner90** [comment]
@fran-worley I just merged 3-x into master. Is there anything from the 2-x branch that is not in 3-x or can I safely delete the 2-x branch?

**Comment by fran-worley** [comment]
Hi @rwwagner90 you're safe to delete 2-x.

I think I still have npm rights if you need me to ship a new release.

probably worth someone with the time to clear out the open pull requests and asking people to rebase and reopen if they want them considered.

**Comment by fran-worley** [comment]
Would also be great to get new docs release out - sadly I have never done this and have no idea how!

**Comment by Gorzas** [comment]
@fran-worley What changes do we need for the docs? Do you mean about changing the handlebars to angle bracket syntax?

**Comment by maxwondercorn** [comment]
@Gorzas Besides syntax changes the dummy app should not migrate from mixins as they don't fit in the Octane world.  We use a base class with inheritance for our tables but there  are other options.  Good discussion [here](https://discuss.emberjs.com/t/best-way-to-replace-mixins/17395) on options

**Comment by Gorzas** [comment]
@maxwondercorn I was working this weekend using Ember v3.24 and the latest version of `ember-light-table`. I'm more inclined to use native classes as a solution, but I'm not sure if it's the best solution.

**Comment by Gorzas** [comment]
I see @maxwondercorn, I really don't know what's the best alternative to mixins. Right now, I'm helping in whatever I can. If any of you could take a look to [this issue](https://github.com/adopted-ember-addons/ember-light-table/issues/758), I'll be happy to continue updating the blueprints :smile:.

**Comment by maxwondercorn** [comment]
@fran-worley @Gorzas 

I realize everyone is in a different place, has moved on or is doing something else since the pandemic 😱  so I'd like to start working on moving ELT forward again. 

I have some bandwidth now and already created a PR #775 to merge your post `2..0.0-beta.5` work   (`origin/fw/update-dummy-app` and `3.0.0-beta.0`) into master.  Apologies in advance...

Per the conversation in this issue, I think there shouldn't be any more effort on a v2 release.  

While the `beta.5` doesn't have updated documentation and some issues (with work arounds), it does work in at least Ember source 3.26 and is embroider compatible.  The first Ember 4 LTS version is near and we should target v3 for compatibility.

For completeness maybe release `v2.0.0-beta.5` as v2 with the caveat that v3 is being worked on.  🤷 After #775 is merged, I can work on getting another v3 beta release.

cc: @rwwagner90

**Comment by Gorzas** [comment]
@maxwondercorn I still work with Ember framework but I'm not currently working with `ember-light-table` library. Anyway, if you need any help, I can take a look. Is there anything I could be testing or any code to review?

**Comment by maxwondercorn** [comment]
@Gorzas Thanks.  IIRC you couldn't get all tests passing on 3.12, so minimum supported would be Ember source 3.13?  My testing confirmed that but wanted to verify it.

3.13 would also be required for converted to colocated Glimmer components.  ATM, nothing to review but once this get's merged I'm sure there will be something 😄
